std/stdio.d

```diff
@@ -651,15 +652,15 @@ Detaches from the underlying file. If the sole owner, calls $(D close).
 
 Throws: $(D ErrnoException) on failure if closing the file.
   */
-    void detach() @safe
+    void detach() @trusted
     {
         if (!_p) return;
-        if (_p.refs == 1)
+        if (atomicLoad(_p.refs) == 1)
             close();
         else
```

Comment:
No, this is not thread-correct.

Assume 2 threads are detaching. Both execute detach simultaneously, both atomically load 2 as the reference count. Both will simply decrement the reference count, resulting in RC of 0, but no destruction. The resource is then leaked.

The correct way to do this is to atomically decrement, and check for 0.